nss-day-cohort-52/nashville-kennels-pine-bears
# name(s) of caretaker(s) are now displayed for each animal card in the AnimalList component

New file(s): N/A
Modified File(s): `Animal.js` in the `animals` directory

Modifications:
1. Added a `users` state variable on line 21
2. Added a `useEffect` on line 34 to fetch all the users and store them in our users state variable
3. Starting on line 93, I mapped over `currentAnimal.animalCaretakers` as well as used `.filter` on our newly populated users array to find the user(s) related to the `animalsCaretakers` of the `currentAnimal`.  
4. The `.filter` was necessary because we needed to be able to access the `name` property of the caretaker(s).  

Steps to test:
1. While in the terminal, use `git fetch --all` in the root directory of this repo 
2. Checkout out to `al-initialBranch`
3. Serve the react app up in the browser using `npm start` on the root directory of this branch
4. Using the navbar, navigate to the "Animals" component of the application
5. Expand each animal card and you should see the name(s) of the caretaker(s) for that particular animal
6. Verify the correct names are being displayed by navigating to your API and matching up the `userId` and `animalId` of each object in the `animalCaretakers` array with the corresponding primary keys of the objects within the `users` array and `animals` array
